crates/wasm/src/compile.rs

```diff
@@ -87,7 +80,10 @@ pub fn compile(args: JsValue) -> JsValue {
     let crate_id = prepare_crate(&mut context, path);
 
     for dependency in options.optional_dependencies_set {
-        add_noir_lib(&mut context, dependency.as_str());
+        // TODO: This doesn't seem like it ever handled a full dependency graph; only a flat list of dependencies
```

Comment:
afaik the old logic added each new crate as a dependency for every crate before it so as long as you provided your list of dependencies in the correct order then you could have a full dependency graph. This isn't possible with the new code.

Reply:
```suggestion
        // TODO: This doesn't handle a full dependency graph; only a flat list of dependencies. To be solved with noirc changes
```

Reply:
I can directly implement the logic that was `propagate_dep` if we want to keep the incorrect behavior mentioned above.

Reply:
Yeah, it's definitely suboptimal as we're attaching dependencies to crates which don't depend on them. It's probably best to have a well defined but restricted behaviour (only one layer of deps) here for the time being rather than something janky.

Reply:
Sounds good. I believe that change makes this PR technically breaking. I'll update and we can hold this until we're done releasing 0.10.x changes.

Reply:
Yep, happy to hold this until the next breaking release or we can maintain current behaviour and split this change into a separate PR.

Reply:
Fair enough, then we'd have a clear changelog message. Give me a few minutes